### PR TITLE
Added AvaJS test runner support

### PR DESCRIPTION
**Pull request by LoveSponge**
### Description

Added AVAJS support
tested on my local machine with AvaJS v3.5.0

Hope it's okay, it's very basic copy/paste from existing functionality - Debugging currently untested/ not working.

### CODE REVIEW

**Comment by felixjb** [comment]
Hi @LoveSponge , I appreciate so much that you opened a pull request. Thank you!

I have not use AvaJS yet, so it would be best if you could bring the new test runner support fully working. This means the debugging functionality.

Also, it would be necessary to update the docs and the configuration options just in case Ava works differently from the others.

Feel free to ask for help and pair programming if you need to.

File: src/runners/AvaTestRunner.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    terminal.sendText(command, true);
+    terminal.show(true);
+  }
+
+  public debugTest(
+    rootPath: WorkspaceFolder,
+    fileName: string,
+    testName: string
+  ) {
+    // const additionalArguments = this.configurationProvider.additionalArguments;
```
Comment (felixjb) [inline src/runners/AvaTestRunner.ts]: :x: unnecessary comments

File: src/runners/AvaTestRunner.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      .environmentVariables;
+    const skipFiles = this.configurationProvider.skipFiles;
+
+    debug.startDebugging(rootPath, {
+      args: [
+        this.transformFileName(fileName),
+        `--debug`,
+        testName
+        // "--runInBand",
+        // ...additionalArguments.split(" ")
```
Comment (felixjb) [inline src/runners/AvaTestRunner.ts]: :x: unnecessary comments

**Comment by LoveSponge** [comment]
@felixjb

Debugging is now working and I have ammended the comments as per your comments.

**Comment by LoveSponge** [comment]
Apologies for skipping over your comment about adding documentation;
I believe the setup is exactly the same, maybe just a link to the AvaJS CLI documentation repository would suffice?
Cheers, hope it's useful to someone.

**Comment by LoveSponge** [comment]
Please also be advised that choosing debugging cannot specify a specific test to debug, it will run all tests in a file, in order.

**Comment by felixjb** [comment]
@LoveSponge Well, the documentation would need at least the minimum to work with the new framework, like it was done with the other ones.

The debugging should actually work for each test individually. If it cannot work this way, it should create codelens only once for debugging with AvaJS.

Also, I would like to say that I am working on some changes, so if and when we merge this pull request, it might still take a while to be publish, as I will need to adapt the code I am working on right now.

**Comment by LoveSponge** [comment]
@felixjb 
Sure thing I'll see to getting it updated.

Unfortunately there's no mention of being able to use '--match' in debug mode for avajs. I have raised a question on their repo.

Okay that's not a problem for me personally - just wanted this change, realised I could do it myself, thought I'd make a pull request because sharing is caring :smile:

Ps. Are the changes you're working on related to adding support for vscode test ui???? :grin:

**Comment by felixjb** [comment]
@LoveSponge I appreciate this so much! Thank you! I use this project myself every single day and want to do so much with it, but I always end up doing something else that is priority at the moment. I would love to have more contributions.

Well, I am working on a few bug fixes, adding snippets for the tests in the frameworks we have now, keybindings will come next. I would love to have a tab on the sideboard to show the stack of tests in the current file and stuff like that, but that takes a lot more effort. Also, a watch option would be awesome, but that takes a lot of effort too.

My priority will be refactoring the codebase so other people can contribute easily. One way of doing that is adding more tests.

**Comment by felixjb** [comment]
@LoveSponge Do you have any suggestions?

**Comment by LoveSponge** [comment]
@felixjb
It's my pleasure, I had this extension before when it was maintained by Geoffrey, I tried to add avajs support back then too but didn't have the knowledge to.

I personally would argue that having a visual list of tests would be a large change with a large payoff in terms of usability. I will see if I can find time to take a look at this.

I don't see the need for a refactoring, I found it easy enough to add support for a whole new test runner... But I won't stand in your way :grin:

Thanks for picking up this project, it's the most lightweight solution there is for what it does, keep up the awesome work!

**Comment by LoveSponge** [comment]
Just to keep this thread updated, I have had correspondence from the guys at avajs and they say that their '--match' functionality should work for debugging.
So I won't make any updates until we get to the bottom of it on that thread.

**Comment by felixjb** [comment]
Alright! Keep me posted

**Comment by LoveSponge** [comment]
Hi, sorry it's been a while, the issue i raised on AvaJS repo was not descriptive enough and has been closed without a resolution, i have since commented with more details in hopes they will re-open.

**Comment by felixjb** [comment]
No worries

**Comment by LoveSponge** [comment]
you wont beleive the fix.

**Comment by LoveSponge** [comment]
@felixjb friendly bump

**Comment by felixjb** [comment]
Hi @LoveSponge I'm sorry for taking too long to respond. I'll check on it soon

**Comment by felixjb** [comment]
@LoveSponge I found a bug on the extension, so I'll fix it before merging this PR

**Comment by felixjb** [comment]
@LoveSponge May I ask you one more favor? Could you update the documentation on the [README.md](https://github.com/felixjb/testify/blob/master/README.md) file?

**Comment by LoveSponge** [comment]
I have updated the readme, all i can see that needs amending is the ## Compatibility section to include '- Ava'... let me know if that's what you meant.

**Comment by felixjb** [comment]
Hey @LoveSponge , this looks good! 

Are you sure you don't want to add your name to the contributors list? I can do it for you too, if you want.

Also, I'm going to migrate from Travis to GitHub actions, as I am having trouble with deploying the extensions with the current process. After I do the migration I'll be able to deploy the new version to the marketplace.

**Comment by LoveSponge** [comment]
@felixjb
Thanks, that's very kind of you!

Okay no problem! ☺ 

I'm just in the process of tidying up the other PR I submitted, which should be done and ready for your review soon;

Cheers

**Comment by LoveSponge** [comment]
@felixjb GH is complaining that some of the commits i've done are not verified - i'm not sure why this is but i can confirm all the commits in this PR have come from me and my work machines.
I have looked up the documentation but haven't found how to verify an already pushed commit... if you could advise on what to do next, that would be helpful.

Cheers!
LS

**Comment by felixjb** [comment]
Hi @LoveSponge ! That's a rule I enabled when I first configured the repository, but I guess it's not really necessary. I will disable it.

**Pull request merged**